src/nexus/check.cc

```diff
@@ -17,4 +18,8 @@ void nx::detail::report_failed_check(nx::detail::check_result const& r, const ch
     }
 
     // TODO: delete r.lhs and r.rhs
+
+    // TODO: make custom exception
+    if (terminate)
+        throw std::logic_error("test assertion failed");
```

Comment:
How about `CC_RUNTIME_ASSERT`? :D

Reply:
I need a custom exception inside nexus that can be caught and is unlikely to be used outside. Additionally, an assertion handler is set to throw exactly that exception. That way nexus can detect and react properly to failed exceptions (there are multiple modes, especially when minimizing property/mct tests).